### PR TITLE
student dashboard

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
default view 

screen shot 2018-04-26 at 2 57 58 pm

after selecting 2nd card (and IAB). Moving to next line is by design

screen shot 2018-04-26 at 2 58 05 pm

ICA view 

screen shot 2018-04-26 at 2 58 16 pm

Claim Scores

screen shot 2018-04-26 at 2 58 22 pm

### CODE REVIEW

File: webapp/src/main/webapp/src/app/dashboard/student-dashboard/student-assessment-card.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="card assessment-aggregate" [ngClass]="{'selected': latestExam.selected}" (click)="selectCard()">
+
+  <span class="check"><i class="fa fa-check"></i></span>
+  <span
+    class="label" [ngClass]="getGradeColor()">{{ 'common.assessment-grade-short-label.' + latestExam.assessment.grade | translate}}</span>
+  <span class="title">{{ latestExam.assessment.label }}</span>
+  <span class="date">{{ date | date }}</span>
+  <p></p>
+  <span class="distribution" [ngClass]="colorService.getPerformanceLevelColorsByAssessmentTypeCode('iab', level)">{{ 'common.assessment-type.iab.performance-level.' + level + '.name' | translate }}</span>
+  <span class="center-block mt-md" style="text-align: center; font-weight: 600">{{ resultCount }} results</span>
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/dashboard/student-dashboard/student-assessment-card.component.html]: oops...styling is here. I'll move this out

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-table.component.ts [hunk 2]
```diff
@@ -117,6 +76,97 @@ export class StudentHistoryTableComponent implements OnInit {
     }
     return builder.build();
   }
+
+  get columns(): Column[] {
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-table.component.ts]: `get actions()` and `get columns()` was necessary for getting the correct behavior when selecting an ICA vs an IAB and getting the table to appear correctly afterward

File: webapp/src/main/webapp/src/styles.less [hunk 3]
```diff
@@ -1129,12 +1129,16 @@ distractor-analysis .ui-widget.ui-table table .ui-table-tbody td.green {
   grid-template-columns: repeat(auto-fit, @spacer * 18);
 }
 
+.card-container.selected {
+  grid-column: 1;
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/styles.less]: this moves the selected table and card to the beginning of a row. This is necessary to account for resized windows

File: webapp/src/main/webapp/src/styles.less [hunk 4]
```diff
@@ -1129,12 +1129,16 @@ distractor-analysis .ui-widget.ui-table table .ui-table-tbody td.green {
...
+}
+
 .card {
   position: relative;
   background-color: @white;
   border-width: @spacer / 4;
   border-style: solid;
-  border-color: transparent;
+  border-color: @gray-lightest;
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/styles.less]: changed to a `gray-lightest` background color to avoid issues with a white card in a white well

**Comment by dsnyder-fairway** [comment]
@jtreuting note we still don't have anything to highlight an ICA/Summative card from IABs other than the card title stating that its an ICA

File: webapp/src/main/webapp/src/app/app.routes.ts [hunk 5]
```diff
@@ -145,6 +145,22 @@ const adminRoute = {
   ]
 };
 
+const studentTestExamHistoryChildRoute = {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/app.routes.ts]: TextExam? ; j

File: webapp/src/main/webapp/src/app/dashboard/group-dashboard/group-assessment-card.component.ts [hunk 6]
```diff
@@ -36,6 +36,10 @@ export class AssessmentCardComponent implements OnInit {
     this.dataWidths[ 2 ] = 100 - (this.dataWidths[ 0 ] + this.dataWidths[ 1 ]);
   }
 
+  get date(): Date {
+    return this.measuredAssessment.date;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/dashboard/group-dashboard/group-assessment-card.component.ts]: why the shortcut just for date?

File: webapp/src/main/webapp/src/app/dashboard/student-dashboard/student-assessment-card.component.html (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/dashboard/student-dashboard/student-assessment-card.component.html]: should these all be <div> ? they are all intended to display on a new line right? also what is the empty `<p>` for?

File: webapp/src/main/webapp/src/app/shared/support/support.ts [hunk 7]
```diff
@@ -237,4 +237,16 @@ export class Utils {
...
     return str.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase();
   }
+
+  /**
+   * Insert a value into an array if the array does not contain that value
+   *
+   * @param {any[]} array the array we are potentially inserting into
+   * @param value the value we wish to possibly insert
+   */
+  static insertIfNotPresent(array: any[], value: any): void {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/support/support.ts]: -> `pushIfAbsent`

Comment (esotrope) [reply webapp/src/main/webapp/src/app/shared/support/support.ts]: also might want to check if the lodash lib already has this

File: webapp/src/main/webapp/src/styles.less [hunk 8]
```diff
@@ -1214,6 +1218,23 @@ distractor-analysis .ui-widget.ui-table table .ui-table-tbody td.green {
...
     }
   }
+
+  .distribution {
+    .btn;
+    width: 100%;
+    pointer-events: none;
+    color: @white;
+    &.blue-dark.aqua {
+      .textured-blue;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/styles.less]: would suggest we change the component to use `.textured-blue` i think we are trying to phase out using `.blue-dark.aqua` right @wtritch ?

**Pull request merged**